Azure/sonic-utilities
# Adding show ip/v6 route summary on Multi-ASIC platforms (master)

**- What I did**
Added Multi-ASIC support of handling "show ip/v6 route summary"
This was not supported for multi-ASIC platform previously.
When running this cmd under multi-ASIC platform the following output are expected:
```
admin@SONiC:$ show ip route summary -d all
asic0:
Route Source         Routes               FIB  (vrf default)
kernel               1                    1
connected            6                    6
static               1                    0
ebgp                 6371                 6371
ibgp                 88                   88
------
Totals               6467                 6466


asic1:
Route Source         Routes               FIB  (vrf default)
kernel               1                    1
connected            6                    6
static               1                    0
ebgp                 6371                 6371
ibgp                 88                   88
------
Totals               6467                 6466


asic2:
Route Source         Routes               FIB  (vrf default)
kernel               1                    1
connected            14                   14
static               1                    0
ebgp                 43                   43
ibgp                 6408                 6408
------
Totals               6467                 6466


asic3:
Route Source         Routes               FIB  (vrf default)
kernel               1                    1
connected            14                   14
static               1                    0
ebgp                 42                   42
ibgp                 6409                 6409
------
Totals               6467                 6466


asic4:
Route Source         Routes               FIB  (vrf default)
kernel               1                    1
connected            6                    6
static               1                    0
ebgp                 0                    0
ibgp                 6462                 6458
------
Totals               6470                 6465


asic5:
Route Source         Routes               FIB  (vrf default)
kernel               1                    1
connected            6                    6
static               1                    0
ebgp                 0                    0
ibgp                 6462                 6458
------
Totals               6470                 6465


admin@SONiC:$ show ip route summary -n asic2
asic2:
Route Source         Routes               FIB  (vrf default)
kernel               1                    1
connected            14                   14
static               1                    0
ebgp                 43                   43
ibgp                 6408                 6408
------
Totals               6467                 6466


admin@SONiC:$ show ip route summary
asic0:
Route Source         Routes               FIB  (vrf default)
kernel               1                    1
connected            6                    6
static               1                    0
ebgp                 6371                 6371
ibgp                 88                   88
------
Totals               6467                 6466


asic1:
Route Source         Routes               FIB  (vrf default)
kernel               1                    1
connected            6                    6
static               1                    0
ebgp                 6371                 6371
ibgp                 88                   88
------
Totals               6467                 6466


asic2:
Route Source         Routes               FIB  (vrf default)
kernel               1                    1
connected            14                   14
static               1                    0
ebgp                 43                   43
ibgp                 6408                 6408
------
Totals               6467                 6466


asic3:
Route Source         Routes               FIB  (vrf default)
kernel               1                    1
connected            14                   14
static               1                    0
ebgp                 42                   42
ibgp                 6409                 6409
------
Totals               6467                 6466


admin@SONiC:$
 
```


**- How to verify it**
Perform "show ip/v6 route summary" on multi-asic platform to see that this option is now supported.
Before the change:
```
admin@SONiC:~$ show ip route summary
% Unknown command: show ip route sum 
```

After the added support:
```
admin@SONiC:~$ show ip route summary
asic0:
Route Source         Routes               FIB  (vrf default)
kernel               1                    1
connected            6                    6
static               1                    0
ebgp                 6371                 6371
ibgp                 88                   88
------
Totals               6467                 6466


asic1:
Route Source         Routes               FIB  (vrf default)
kernel               1                    1
connected            6                    6
static               1                    0
ebgp                 6371                 6371
ibgp                 88                   88
------
Totals               6467                 6466


asic2:
Route Source         Routes               FIB  (vrf default)
kernel               1                    1
connected            14                   14
static               1                    0
ebgp                 43                   43
ibgp                 6408                 6408
------
Totals               6467                 6466


asic3:
Route Source         Routes               FIB  (vrf default)
kernel               1                    1
connected            14                   14
static               1                    0
ebgp                 42                   42
ibgp                 6409                 6409
------
Totals               6467                 6466


admin@SONiC:$
```
